runelite-client/src/main/java/net/runelite/client/plugins/hunter/HunterPlugin.java

```diff
@@ -377,6 +393,76 @@ else if (containsBoulder) // For traps like deadfalls. This is different because
 		lastTickLocalPlayerLocation = client.getLocalPlayer().getWorldLocation();
 	}
 
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged interactingChanged)
+	{
+		mapActorToChinTarget(interactingChanged.getTarget(), interactingChanged.getSource());
+	}
+
+	/**
+	 *	 				DEAD TRAP MECHANICS
+	 *	 When spawning a trap, an invisible null NPC is spawned at the same location.
+	 *	 This "null" npc handles all the interactions between a trap and a target.
+	 * 	 If two hunter traps target the same Actor, then the trap
+	 * 	 which did not catch the target will get stuck in a "dead" state.
+	 * 	 It will no longer be able to re-target another actor.
+	 * 	 Therefore, the user can remove these traps and re-place them.
+	 *
+	 *	 The code below keeps track of which traps interacted with every
+	 *	 chinchompa. If any traps are left over in this list when
+	 *	 the chinchompa is caught, the trap enters a dead state.
+	 */
+	private void mapActorToChinTarget(Actor target, Actor source)
+	{
+		if (target != null && source instanceof NPC)
+		{
+			// the trap is a multinpc, therefore the ID must be retrieved from the untransformed composition or it will be -1
+			if (((NPC) source).getComposition().getId() == NULL_8740)
```

Comment:
More conditional combining!
```suggestion
		// the trap is a multinpc, therefore the ID must be retrieved from the untransformed composition or it will be -1
		if (target != null && source instanceof NPC && ((NPC) source).getComposition().getId() == NULL_8740)
		{
```